dts/bindings/rtc/nxp,irtc.yaml

```diff
@@ -0,0 +1,37 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: IRTC
+
+compatible: "nxp,irtc"
+
+include: rtc.yaml
+
+properties:
+  reg:
+    required: true
+
+  clock-select:
+    type: int
+    required: true
+    description: |
+      The input clock source for IRTC.
+      0 <- 16k
+      1 <- 32k
+
+  alarms-count:
+    type: int
+    required: true
```

Comment:
This value should be inherited from rtc-device.yaml :) where it is not required, and has the default 0. The default is there to allow the application to check the number of alarms supported by an RTC at compile time.

Reply:
@bjarki-andreasen maybe I don't understand, if this is supposed to describe a property of hardware, why shouldn't it be required: true? If it's required, then it will always have a value to check by the application still, even if it is specified in DT as 0 still.

Reply:
For historical reasons, RTCs used the counter API, and already had bindings. These original bindings did not contain the alarms-count property. To avoid breaking downstream users, a default was added allowing the same bindings to be used for counters and RTCs. It should be required at some point, and would have been from the beginning if there weren't already conflicting bindings in tree :)

Reply:
I guess it makes sense, it would be a lot of counters to update

Reply:
@EmilioCBen this property should not be being overridden here, please remove it from this file

Reply:
Updated, added rtc-device.yaml